Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/messaging/service/decorator/HttpHeaderServiceEndpointDecoratorTest.java

```diff
@@ -42,106 +47,125 @@
-        
-        headerDecorator.configure();
-        
-        validateConfiguration(headerDecorator);
-        BindingProvider bp = (BindingProvider) headerDecorator.getPort();
-        Map<String, List<String>> bpMap = (Map<String, List<String>>) bp.getRequestContext().get(NhincConstants.CUSTOM_HTTP_HEADERS);
-        assert(bpMap.containsKey("customName"));
-    }
-    
-    private void validateConfiguration(HttpHeaderServiceEndpointDecorator headerDecorator) {
-        assertEquals(headerDecorator.getHTTPClientPolicy().getConnection(), ConnectionType.KEEP_ALIVE);
-        
-        BindingProvider bp = (BindingProvider) headerDecorator.getPort();
-        Map<String, List<String>> bpMap = (Map<String, List<String>>) bp.getRequestContext().get(NhincConstants.CUSTOM_HTTP_HEADERS);
-        assertNotNull(bpMap);
-        assertEquals(bpMap.size(), 1);
-    }
+	public HttpHeaderServiceEndpointDecoratorTest() {
+	}
+
+	@Before
+	public void setUp() {
+
+	}
+
+	@After
+	public void tearDown() {
+	}
+
+	@Test
+	public void testConfigure_WithAssertionValues() throws PropertyAccessException {
+		CONNECTTestClient<TestServicePortType> testClient = new CONNECTTestClient<>(new TestServicePortDescriptor());
+
+		ServiceEndpoint<TestServicePortType> serviceEndpoint = testClient.getServiceEndpoint();
+
+		final PropertyAccessor mockPropAccessor = mock(PropertyAccessor.class);
+
+		AssertionType assertion = new AssertionType();
+		assertion.setKeepAlive("true");
+
+		CONNECTCustomHttpHeadersType assertionHeaders = new CONNECTCustomHttpHeadersType();
+		assertionHeaders.setHeaderName("headerName");
+		assertionHeaders.setHeaderValue("headerValue");
+
+		assertion.getCONNECTCustomHttpHeaders().add(assertionHeaders);
+
+		HttpHeaderServiceEndpointDecorator headerDecorator = new HttpHeaderServiceEndpointDecorator(serviceEndpoint,
+				assertion) {
+
+			@Override
+			protected PropertyAccessor getPropertyAccessor() {
+				return mockPropAccessor;
+			}
+		};
+
+		when(mockPropAccessor.getPropertyNames(NhincConstants.GATEWAY_PROPERTY_FILE)).thenReturn(new HashSet<String>());
+
+		headerDecorator.configure();
+
+		validateConfiguration(headerDecorator);
+	}
+
+	@Test
+	public void testConfigure_WithoutValues() throws PropertyAccessException {
+		CONNECTTestClient<TestServicePortType> testClient = new CONNECTTestClient<>(new TestServicePortDescriptor());
+
+		ServiceEndpoint<TestServicePortType> serviceEndpoint = testClient.getServiceEndpoint();
+
+		AssertionType assertion = new AssertionType();
+
+		HttpHeaderServiceEndpointDecorator headerDecorator = new HttpHeaderServiceEndpointDecorator(serviceEndpoint,
+				assertion);
+
+		headerDecorator.configure();
+
+		BindingProvider bp = (BindingProvider) headerDecorator.getPort();
+		Map<String, List<String>> bpMap = (Map<String, List<String>>) bp.getRequestContext()
+				.get(NhincConstants.CUSTOM_HTTP_HEADERS);
+		assert bpMap.containsKey("customName");
```

Comment:
use assertion from junit library. Don't use out of the box assert.  It turns off by default and give u false positive.
 (http://docs.oracle.com/javase/7/docs/technotes/guides/language/assert.html#usage-conditions).  To turn on that feature, you need to add -ea to vm argument which we don't want to do.  Recommend to replace with
Assert.assertNotNull(bpMap);
Assert.assertTrue(bpMap.containsKey("customName"));